contractcourt/contract_resolvers.go

```diff
@@ -46,11 +46,6 @@ type ContractResolver interface {
 	// passed Writer.
 	Encode(w io.Writer) error
 
-	// Decode attempts to decode an encoded ContractResolver from the
-	// passed Reader instance, returning an active ContractResolver
-	// instance.
-	Decode(r io.Reader) error
```

Comment:
it is a bit weird to have an `Encode` method, but no `Decode`. I think we should keep it, if you need to make it used you can call it outside the switch statement where the resolver type is checked.

Reply:
See comment about encapsulation

Reply:
Also, this method was never called on the `ContractResolver` interface. This commit builds